lib/resources/nginx_module.rb

```diff
@@ -0,0 +1,35 @@
+# encoding: utf-8
+# author: Rony Xavier, rx294@gmail.com
+# author: Aaron Lippold, lippold@gmail.com
+
+module Inspec::Resources
+  class NginxModule < Inspec.resource(1)
+    name 'nginx_module'
+    desc 'Use the nginx_module InSpec audit resource to test nginx modules on Linux platforms.'
+    example "
+      describe nginx_module('http_auth_request') do
+        it { should be_loaded }
+      end
+      "
+
+    def initialize(modulename = nil, nginx_path = nil)
```

Comment:
Because these aren't named options, it creates two weird UX issues:

1. It's not as declarative, so if a path is given, you don't know what it's for...
1. If someone wants to list all modules but needs to give a path, they need to pass a `nil` as the first argument

This would probably be better using an options hash:

```
def initialize(opts={})
  @nginx_module = opts[:module_name]
  @nginx_path = opts[:nginx_path]
   # etc...
end
```

The user would then use this resource like this:

```
describe nginx_module do
  its('loaded_modules.length') { should cmp > 1 }
end

describe nginx_module(nginx_path: '/opt/mycompany/bin/nginx') do
  its('loaded_modules') { should include 'foo' }
end

describe nginx_module(module_name: 'my_module') do
  it { should be_loaded }
end
```

You'll obviously need to adjust your tests as well. 😄